x/tss/keeper/genesis.go

```diff
@@ -0,0 +1,112 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/x/tss/exported"
+	"github.com/axelarnetwork/axelar-core/x/tss/types"
+)
+
+// InitGenesis initializes the reward module's state from a given genesis state.
+func (k Keeper) InitGenesis(ctx sdk.Context, snapshotter types.Snapshotter, genState *types.GenesisState) {
+	k.SetParams(ctx, genState.Params)
+
+	rotationCountMap := make(map[string]map[exported.KeyRole]int64)
+	for _, key := range genState.Keys {
+		if _, ok := k.GetKey(ctx, key.ID); ok {
+			panic(fmt.Errorf("key %s already set", key.ID))
+		}
+
+		k.setKey(ctx, key)
+
+		if key.RotationCount > 0 {
+			if _, ok := k.getKeyID(ctx, key.Chain, key.RotationCount, key.Role); ok {
+				panic(fmt.Errorf("key ID for chain %s, rotation count %d and role %s already set", key.Chain, key.RotationCount, key.Role.SimpleString()))
+			}
+
+			k.setKeyID(ctx, key.Chain, key.RotationCount, key.Role, key.ID)
+		}
+
+		if key.Role != exported.ExternalKey {
+			if _, ok := snapshotter.GetSnapshot(ctx, key.SnapshotCounter); !ok {
```

Comment:
Seems like we need to have the snapshooter ready by the time we initialize from genesis, which is enabled by the fact that the tss keeper knows about the snapshotter. But the snapshotter also knows about the tss keeper, and it might also need to access the tss keeper upon initialization. So I wonder if we have here a circular dependency, in the sense that tss needs to have the snapshotter ready, but so does the snapshotter need tss to be ready?

Reply:
No, keeper initialization and genesis initialization are different and separated.

Reply:
Sure, I just wonder if genesis initialization will have a circular dependency